src/eradiate/radprops/_array.py

```diff
@@ -0,0 +1,254 @@
+    collision coefficients).
+
+    The array radiative profile is built from absorption and scattering
+    coefficient data provided as input. This can be useful for debugging and
+    benchmarking.
+    """
+
+    sigma_a: xr.DataArray | None = documented(
+        attrs.field(
+            default=None,
+        ),
+        doc="``DataArray`` of absorption coefficients "
+        "The ``DataArray`` is composed of two dimensions ``{w, z}`` representing the "
+        "wavelength and altitude respectively. Note that ``w`` must be of length "
+        "2 minimum to be correctly interpolated. ",
+        type="DataArray or None",
+        init_type="DataArray or None",
+        default="None",
+    )
+
+    sigma_s: xr.DataArray | None = documented(
+        attrs.field(
+            default=None,
+        ),
+        doc="``DataArray`` of scattering coefficients "
+        "The ``DataArray`` is composed of two dimensions ``{w, z}`` representing the "
+        "wavelength and altitude respectively. Note that ``w`` must be of length "
+        "2 minimum to be correctly interpolated. ",
+        type="DataArray or None",
+        init_type="DataArray or None",
+        default="None",
+    )
+
+    has_absorption: bool = documented(
+        attrs.field(
+            default=True,
+            converter=bool,
+            validator=attrs.validators.instance_of(bool),
+        ),
+        doc="Absorption switch. If ``True``, the absorption coefficient is "
+        "computed. Else, the absorption coefficient is not computed and "
+        "instead set to zero.",
+        type="bool",
+        default="True",
+    )
+
+    has_scattering: bool = documented(
+        attrs.field(
+            default=True,
+            converter=bool,
+            validator=attrs.validators.instance_of(bool),
+        ),
+        doc="Scattering switch. If ``True``, the scattering coefficient is "
+        "computed. Else, the scattering coefficient is not computed and "
+        "instead set to zero.",
+        type="bool",
+        default="True",
+    )
+
+    rayleigh_depolarization: np.ndarray = documented(
+        attrs.field(
+            converter=lambda x: np.array(x, dtype=np.float64),
+            kw_only=True,
+            factory=lambda: np.array(0.0),
+        ),
+        type="ndarray",
+        doc="Depolarization factor of the rayleigh phase function. "
+        "A ``ndarray`` will be interpreted as a description of the depolarization "
+        "factor at different levels of the atmosphere. Must be shaped (N,) with "
+        "N the number of layers.",
+    )
+
+    interpolation_method: str = documented(
+        attrs.field(
+            default="nearest",
+            converter=str,
+            validator=attrs.validators.instance_of(str),
+        ),
+        doc="Method of interpolation of the absorption and scattering coefficients. ",
+        type="str",
+        default="nearest",
```

Comment:
Which methods do we support? I think making the allowed options explicit would prevent unpleasant mistakes. That would imply changing the type to something like `Literal["nearest", "linear"]` and the `init_type` doc field to something like `{"nearest", "linear"}`. The validator can then be `attrs.validators.in_`.